File: Part-3/PART3/Frontend/src/services/youtubeservice.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8002/api/youtube';

export const fetchYouTubeVideos = async (query) => {
  try {
    const response = await axios.get(`${API_BASE_URL}/search`, {
      params: {
        query,
        max_results: 6
      }
    });

    return response.data.videos;
  } catch (error) {
    console.error('Error fetching YouTube videos:', error);
    return [];
  }
};

File: Part-3/PART3/Backend/services/youtube_service.py
import os
import requests
from dotenv import load_dotenv

# Load variables from .env file
load_dotenv()

YOUTUBE_API_KEY = os.getenv("YOUTUBE_API_KEY")
BASE_URL = "https://www.googleapis.com/youtube/v3/search"

def search_youtube_videos(query: str, max_results: int = 10):
    if not YOUTUBE_API_KEY:
        return {"error": "YOUTUBE_API_KEY is not set in the environment variables."}
    
    params = {
        "part": "snippet",
        "q": query,
        "type": "video",
        "maxResults": max_results,
        "key": YOUTUBE_API_KEY
    }
    
    try:
        response = requests.get(BASE_URL, params=params)
        response.raise_for_status()
        data = response.json()
        
        videos = []
        for item in data.get("items", []):
            videos.append({
                "videoId": item["id"]["videoId"],
                "title": item["snippet"]["title"],
                "description": item["snippet"]["description"],
                "thumbnail": item["snippet"]["thumbnails"]["medium"]["url"],
                "channelTitle": item["snippet"]["channelTitle"]
            })
        return videos
    except requests.exceptions.RequestException as e:
        return {"error": f"Failed to fetch videos: {str(e)}"}